# Estimation And Confidence Intervals

## --> Import dependencies

In [218]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [219]:
data=[1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]
df=pd.DataFrame(data,index=range(1,len(data)+1),columns=['Print-heads'])

In [220]:
df

Print-heads
1          1.13
2          1.55
3          1.43
4          0.92
5          1.25
6          1.36
7          1.32
8          0.85
9          1.07
10         1.48
11         1.20
12         1.33
13         1.18
14         1.22
15         1.29

# --> Task A

## Build 99% Confidence Interval Using Sample Standard Deviation

In [223]:
df

Print-heads
1          1.13
2          1.55
3          1.43
4          0.92
5          1.25
6          1.36
7          1.32
8          0.85
9          1.07
10         1.48
11         1.20
12         1.33
13         1.18
14         1.22
15         1.29

In [224]:
m=df['Print-heads'].mean()

In [225]:
std=df['Print-heads'].std()

In [296]:
df2=14
ci=0.99
q=(1+ci)/2

In [298]:
from scipy import stats

In [300]:
t_c=stats.t.ppf(q,df2)

In [302]:
t_c

2.976842734370834

In [304]:
margin_of_error=t_c*std/np.sqrt(len(df))

In [306]:
margin_of_error

0.14846932822817593

In [308]:
CI=(round(m-margin_of_error,2),round(m+margin_of_error,2))

In [310]:
CI

(1.09, 1.39)

### In this scenario, the print heads durability in terms of the number of characters before failure is between 1.09 to 1.39 characters (in millions)

# --> Task B

## Build 99% Confidence Interval Using Sample Standard Deviation

In [315]:
df

Print-heads
1          1.13
2          1.55
3          1.43
4          0.92
5          1.25
6          1.36
7          1.32
8          0.85
9          1.07
10         1.48
11         1.20
12         1.33
13         1.18
14         1.22
15         1.29

In [317]:
std2=0.2

In [319]:
z=(1+0.99)/2

In [321]:
z

0.995

In [323]:
z_c=stats.norm.ppf(q=z)

In [325]:
z_c

2.5758293035489004

In [330]:
margin_of_error2=z_c*std2/np.sqrt(len(df))

In [332]:
CI2=(round(m-margin_of_error2,2),round(m+margin_of_error2,2))

In [334]:
CI2

(1.11, 1.37)

### In this scenario, the print heads durability in terms of the number of characters before failure is between 1.11 to 1.37 characters (in millions)